iquip/apps/stage.py

```diff
@@ -0,0 +1,163 @@
+"""Module for stage controller."""
+
+import functools
+import logging
+from typing import Callable, Dict, Optional, Tuple
+
+from sipyco.pc_rpc import Client
+from PyQt5.QtCore import QObject, Qt, pyqtSignal, pyqtSlot
+
+logger = logging.getLogger(__name__)
+
+RPCTargetInfo = Tuple[str, int, str]  # ip, port, target_name
+
+def use_client(function: Callable[..., None]) -> Callable[..., None]:
+    """Decorator which substitutes a string key to a client object.
+
+    If the key does not exist, function is not called at all.
+    If an OSError occurs while running function, the RPC client is closed and
+        removed from the client dictionary. 
+    
+    Args:
+        function: Decorated function. It should take a Client object as the
+            first argument.
+    """
+    @functools.wraps(function)
+    def wrapped(self: "StageManager", key: str, *args, **kwargs):
```

Comment:
Referred to [stackoverflow](https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class), and [PEP 563](https://peps.python.org/pep-0563/#enabling-the-future-behavior-in-python-3-7), you might be able to use `__future__` module.

Please check it!

Reply:
Good to know! Thank you, I will apply it.